src/utils/signedUrlReplacer.ts

```diff
@@ -32,3 +35,31 @@ export async function getFilePathFromUrl(url: string) {
     return null
   }
 }
+
+export async function updateTaskIdOfScrapImagesAfterCreation(
+  db: PrismaClient<Prisma.PrismaClientOptions, never, DefaultArgs>,
+  htmlString: string,
+  task_id: string,
+) {
+  const imgTagRegex = /<img\s+[^>]*src="([^"]+)"[^>]*>/g //expression used to match all img tags in provided HTML string.
+  let match
+  const filePaths: string[] = []
+  while ((match = imgTagRegex.exec(htmlString)) !== null) {
+    const originalSrc = match[1]
+    const filePath = await getFilePathFromUrl(originalSrc)
+    if (filePath) {
+      filePaths.push(filePath)
+    }
+  }
+
+  await db.scrapImage.updateMany({
+    where: {
+      filePath: {
+        in: filePaths,
+      },
+    },
+    data: {
+      taskId: task_id,
+    },
+  })
+}
```

Comment:
This is better located in ScrapImagesService due to its relevance with that domain

Reply:
I think this is more relevant to creating tasks, since creating scrap images for tasks is also in tasks.service, would it be good to keep this in tasks service?

Reply:
The main resource that is being manipulated is ScrapImage so it makes more sense to keep it in that one